web/index.html

```diff
@@ -0,0 +1,111 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Ferrostar Web Demo</title>
+    <link rel="stylesheet" href="./src/index.css" />
+    <script type="module" src="/src/ferrostar-core.ts"></script>
+    <script type="module" src="/src/location.ts"></script>
+  </head>
+  <body>
+    <ferrostar-core
+      id="core"
+      valhallaEndpointUrl="https://valhalla1.openstreetmap.de/route"
+      profile="bicycle"
+      httpClient="${fetch}"
+      lastLocation="${locationProvider}"
+      costingOptions="${{ bicycle: { use_roads: 0.2 } }}"
+    ></ferrostar-core>
+
+    <p class="controls">
+      <input type="text" id="destination" placeholder="Where do you want to go?" />
+      <span>
+        <button id="start">Start Navigation</button>
+        or
+        <button id="simulate">Simulate Navigation</button>
+      </span>
+    </p>
+
+    <script type="module">
+      import { SimulatedLocationProvider, BrowserLocationProvider } from "/src/location.ts";
+
+      // FIXME: type + should be camelCase
+      const config = {
+        step_advance: {
+          RelativeLineStringDistance: {
+            minimum_horizontal_accuracy: 25,
+            automatic_advance_distance: 10,
+          },
+        },
+        route_deviation_tracking: {
+          StaticThreshold: {
+            minimum_horizontal_accuracy: 25,
+            max_acceptable_deviation: 10.0,
+          },
+        },
+      };
+
+      document.addEventListener("DOMContentLoaded", async () => {
+        const core = document.getElementById("core");
+        const destinationInput = document.getElementById("destination");
+        const startNavigationButton = document.getElementById("start");
+        const simulateNavigationButton = document.getElementById("simulate");
+
+        startNavigationButton.addEventListener("click", async () => {
+          const destination = destinationInput.value;
+
+          const { lat, lon } = await fetch("https://nominatim.openstreetmap.org/search?q=" + destination + "&format=json")
+            .then((response) => response.json())
+            .then((data) => data[0]);
+
+          const waypoints = [{ coordinate: { lat: parseFloat(lat), lng: parseFloat(lon) }, kind: "Break" }];
+
+          const locationProvider = new BrowserLocationProvider();
+          locationProvider.requestPermission();
+          locationProvider.start();
+
+          // This approach is not ideal, any better way to wait for the locationProvider to acquire the first location?
+          while (!locationProvider.lastLocation) {
+            await new Promise((resolve) => setTimeout(resolve, 100));
+          }
```

Comment:
Good question; I'll need to review the MDN docs on this.

Reply:
Whoops; I misread this initially. This isn't specific to the geolocation API :D 

For the `BrowserLocationProvider` we can basically just wrap this function which calls back once it gets the location (or errors): https://developer.mozilla.org/en-US/docs/Web/API/Geolocation/getCurrentPosition.

For the `SimulatedLocationProvider` we'd have to mimic the interface ourselves but it shouldn't be too hard.. I think adding a list of waiting callbacks or something that requested the location (slightly different from `updateCallback` which is more of a long-term subscription).